### PR TITLE
Eosc overhaul

### PR DESCRIPTION
**Pull request by nathanhourt**
See discussion in #203

### CODE REVIEW

**Comment by nathanhourt** [comment]
Are there any docs to update other than https://eosio.github.io/eos/group__eosc.html?

**Comment by bytemaster** [comment]
This pull request needs to be rebased on current master with fixed submodule.

File: programs/eosc/main.cpp [hunk 1]
```diff
@@ -21,6 +22,8 @@
 
 #include <fc/io/fstream.hpp>
 
+#include "CLI11.hpp"
```
Comment (heifner) [inline programs/eosc/main.cpp]: Comment with verison of CLI11 used might be helpful, I assume version 1.1.

Comment (nathanhourt) [reply programs/eosc/main.cpp]: At this point we're using the HEAD version, as 1.1 contained a bug [which I reported](https://github.com/CLIUtils/CLI11/issues/23) and he fixed, slated for release in version 1.2. I've included the header in our source distribution so there's no worries about having to go find the right version.

**Pull request merged**